packages/caliper-core/test/manager/caliper-engine.js

```diff
@@ -0,0 +1,139 @@
+        });
+
+        it('should set the workspace and initial return code', function() {
+            // TODO: Implement test
+        });
+    });
+
+    describe('Run Method Execution Flow', function() {
+
+        context('When start commands are to be executed', function() {
+            it('should execute the start command successfully', function() {
+                // TODO: Implement test
+            });
+
+            it('should handle errors during start command execution', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When start commands are to be skipped', function() {
+            it('should skip executing the start command', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When initialization phase is to be performed', function() {
+            it('should initialize the network connector successfully', function() {
+                // TODO: Implement test
+            });
+
+            it('should handle errors during network initialization', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When initialization phase is to be skipped', function() {
+            it('should skip the network initialization phase', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When smart contract installation is to be performed', function() {
+            it('should install the smart contract successfully', function() {
+                // TODO: Implement test
+            });
+
+            it('should handle errors during smart contract installation', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When smart contract installation is to be skipped', function() {
+            it('should skip the smart contract installation phase', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When test phase is to be performed', function() {
+            it('should execute test rounds using the round orchestrator', function() {
+                // TODO: Implement test
+            });
+
+            it('should handle errors during the test phase execution', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When test phase is to be skipped', function() {
+            it('should skip the test phase execution', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When an error occurs during run execution', function() {
+            it('should catch and log the error, setting an appropriate return code', function() {
+                // TODO: Implement test
+            });
+        });
+
+        context('When end commands are to be executed', function() {
+            it('should execute the end command successfully', function() {
```

Comment:
end commands are always run if the flag is set to true, even if an error occurs on any of the other stages. This scenario needs to be tested